src/squlearn/kernel/matrix/kernel_optimizer.py

```diff
@@ -0,0 +1,89 @@
+"""Quantum Kernel Optimizer"""
+
+import numpy as np
+from functools import partial
+from typing import Optional, Sequence
+
+from squlearn.kernel.matrix.kernel_matrix_base import KernelMatrixBase
+from squlearn.kernel.optimization.kernel_loss_base import KernelLossBase
+from squlearn.optimizers.optimizer_base import OptimizerBase
+
+
+class KernelOptimizer(KernelMatrixBase):
+    """
+    Quantum kernel optimizer.
+    This class can be used to optimize the variational parameters of a quantum kernel.
+
+    Args:
+        loss (KernelLossBase): The loss function to be minimized.
+        optimizer (OptimizerBase): The optimizer to be used.
+        quantum_kernel (KernelMatrixBase): The quantum kernel to be optimized.
+        initial_parameters (Optional[Sequence[float]]): Initial parameters for the optimizer.
+    """
+
+    def __init__(
+        self,
+        quantum_kernel: KernelMatrixBase,
+        loss: KernelLossBase = None,
+        optimizer: OptimizerBase = None,
+        initial_parameters: Optional[Sequence[float]] = None,
+    ) -> None:
+        super().__init__(
+            encoding_circuit=quantum_kernel.encoding_circuit,
+            executor=quantum_kernel._executor,
+            initial_parameters=initial_parameters,
+            parameter_seed=quantum_kernel._parameter_seed,
+            regularization=quantum_kernel._regularization,
+        )
+        self._quantum_kernel = quantum_kernel
+        self._loss = loss
+        self._optimizer = optimizer
+        self._initial_parameters = (
+            initial_parameters
+            if initial_parameters is not None
+            else self._quantum_kernel.parameters
+        )
+        self._opt_result = None
+        self._optimal_value = None
+        self._optimal_point = None
```

Comment:
I think a getter for the optimal_point would be great, I would also suggest to rename it into _optimal_parameters